modular_skyrat/code/modules/projectiles/guns/ballistic/revolver.dm

```diff
@@ -60,3 +60,60 @@
 	caliber = "all"
 	ammo_type = /obj/item/ammo_casing
 	max_ammo = 1
+
+
+
+//
+///////////////
+// REVOLVERS //
+///////////////
+//		Revolving rifles! We have three versions. An improvised slower firing one, a normal one, and a golden premium one that fires .45 instead of .38
+//		The gold rifle uses .45, it's only 5 more points of damage unfortunately. Fun hint: A box of .45 bullets functions the same as a speedloader.
+//
+
+/obj/item/gun/ballistic/revolver/rifle
+	name = "\improper .38 Revolving Rifle"
+	desc = "A revolving rifle chambered in .38. "
+	icon = 'modular_skyrat/icons/obj/guns/projectile40x32.dmi'
+	icon_state = "revolving-rifle"
+	item_state = "revolving"
+	mag_type = /obj/item/ammo_box/magazine/internal/cylinder/rev38	//This is just a detective's revolver but it's too big for bags..
+	pixel_x = -4	// It's centred on a 40x32 pixel spritesheet.
+	w_class = WEIGHT_CLASS_BULKY
+	weapon_weight = WEAPON_HEAVY // The entire purpose of this is that it's a bulky rifle instead of a revolver.
+	slot_flags = ITEM_SLOT_BELT
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+	lefthand_file = 'modular_skyrat/icons/mob/inhands/weapons/64x_guns_left.dmi'
+	righthand_file = 'modular_skyrat/icons/mob/inhands/weapons/64x_guns_right.dmi'
+	pixel_x = -4
+
+/obj/item/gun/ballistic/revolver/rifle/improvised
+	name = "\improper Improvised .38 Revolving Rifle"
+	desc = "A crudely made revolving rifle. It fires .38 rounds. The cylinder doesn't rotate very well."
+	icon_state = "revolving-rifle"
+	mag_type = /obj/item/ammo_box/magazine/internal/cylinder/rev38	//TAs far as improvised weapons go, this is fairly decent, this isn't half bad.
```

Comment:
.38 is a very dangerous caliber, namely because of prime target
/obj/item/projectile/bullet/c38/dumdum
high embed, high bleed on embed, you can easily make a loadout that'd make sec blush with
dumdum>match>match>dumdum>dumdum>match

Reply:
I'll have to check the availability of dumdum/match rounds in the autolathe and seclathe.
I don't think you can get them outside of cargo or the uplink but thanks for reminding me.

Reply:
Given it's a trip to cargo for it I'm going to leave it as is. If you're getting match-grade ammunition and dumdum rounds from cargo you're already able to get armoury weapons.
![image](https://user-images.githubusercontent.com/53862927/92124793-22372b00-edf6-11ea-93ed-de57252c8443.png)